### PR TITLE
Resources: Tags as theme filter

### PR DESCRIPTION
**Pull request by davidbgomes**
[Link resource tags to corresponding resources categories](https://app.clickup.com/t/865cjftra)

image

### CODE REVIEW

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

I think here there was a misunderstanding. I think this feature should be:
When you click on a theme on comparative analysis, the app should navigate to comparative analysis item filtering by theme.

You have an example of this behaviour to click on a flag (country analysis) on this page then the app navigates to the country analysis filtering by the selected country.

@saragilcas can you confirm that I'm right?

**Comment by saragilcas** [comment]
Hi, @davidbgomes it is exactly as @xurxodev says, although his example is not working right now. Let me know if you still have doubts,

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

Please apply the last changes in the PR https://github.com/EyeSeeTea/uhcpw/pull/74 here because the feature is equal but for different param:
* Sync URL with theme dropdown
* Cancel request in useEffect (review is working because is the same custom hook)

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

Some issues to review, I've added comments in the code

File: src/webapp/components/resources/common/AnalysisItem.tsx [hunk 1]
```diff
@@ -17,42 +20,55 @@ const themeColor: Record<string, string> = {
...
 
-            <Box height={20} />
-            <Tag theme={analysisResource.theme} color={themeColor[analysisResource.theme]} />
-            {!minimalVersion && (
-                <>
-                    <Row>
-                        <Typography variant="body1">{analysisResource.date.getFullYear()}</Typography>
+        const handleTagClick = () => {
+            filters.theme.onSelectedChange(analysisResource.theme);
+            history.push(`/resources/comparative-analysis?theme=${analysisResource.theme}`);
```
[the remark this answers is not in the document]
Comment (davidbgomes) [reply src/webapp/components/resources/common/AnalysisItem.tsx]: Thank you Jorge, I've fixed that part where clicking on a country-analysis tag leads you to comparative-analysis, we had a meeting and Sara confirmed that the tags on country-analysis should also be a filter, so it's done on both.

**Pull request merged**